### PR TITLE
Issue: #81: 7.1-Listing favorite items from local db.

### PR DESCRIPTION
**Pull request by ptyagicodecamp**
Fetching favorite items from the local database and displaying them in Favorite listing screen.

### CODE REVIEW

**Comment by 4seer** [comment]
Please specify in what file and line do you fetch items from local database. Looks like you are missing some files in commit.

File: lib/data/fake_model/fake_product_repository.dart [hunk 1]
```diff
@@ -149,7 +149,8 @@ class FakeProductRepository extends ProductRepository with FavoritesRepository {
       SortRules sortRules = const SortRules(),
       FilterRules filterRules}) async {
     return (await _getFavorites())
-        .map((value) => FavoriteProduct(value, HashMap()));
+        .map((value) => FavoriteProduct(value, HashMap()))
+        .toList();
```
Comment (ptyagicodecamp) [inline lib/data/fake_model/fake_product_repository.dart:153]: Add `toList()` to return favorite items. Earlier it was not returning empty results.

**Comment by ptyagicodecamp** [comment]
> Please specify in what file and line do you fetch items from local database. Looks like you are missing some files in commit.

Looks like I fixed the bug in displaying favorite items returned from `_getFavorite()` method. I see local db is only being populated for tests, and using fake data for app. Am I missing something?

File: lib/data/fake_model/fake_product_repository.dart (hunk 1 above)
Comment (4seer) [reply lib/data/fake_model/fake_product_repository.dart:153]: ok

**Comment by 4seer** [comment]
> 
> 
> > Please specify in what file and line do you fetch items from local database. Looks like you are missing some files in commit.
> 
> Looks like I fixed the bug in displaying favorite items returned from `_getFavorite()` method. I see local db is only being populated for tests, and using fake data for app. Am I missing something?

Ok. Initial message "Fetching favorite items from the local database and displaying them in Favorite listing screen." was not clear enough and was leaving an impression that some real fetching was supposed to take place.

**Pull request merged**